[CI] OldMappingsIT class failing

**Build Scans:**
- [elasticsearch-periodic #12623 / openjdk26_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/sdjlpjuhbgkrg)
- [elasticsearch-pull-request #168652 / part-2](https://gradle-enterprise.elastic.co/s/4x2nfyxnanq4o)
- [elasticsearch-periodic-platform-support #12581 / almalinux-8-aarch64_checkpart2_platform-support-arm](https://gradle-enterprise.elastic.co/s/a2wfbcmfwwy2i)
- [elasticsearch-pull-request #168463 / part-2](https://gradle-enterprise.elastic.co/s/6kctptw7c653w)
- [elasticsearch-pull-request #168463 / part-2](https://gradle-enterprise.elastic.co/s/74oznozdeeqiq)

**Reproduction Line:**
```
undefined
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.oldrepos.OldMappingsIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('org.elasticsearch.oldrepos.OldMappingsIT'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
undefined
```

**Issue Reasons:**
- [main] 5 failures in class org.elasticsearch.oldrepos.OldMappingsIT (0.6% fail rate in 907 executions)
- [main] 3 failures in step part-2 (2.5% fail rate in 119 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (1.7% fail rate in 115 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-core-infra.

:eyes:

## Test Failure Analysis

**Summary**
`OldMappingsIT.testSurvivesRestart` intermittently fails (~0.6–2.5% of runs) with:
```
java.lang.RuntimeException: An error occurred while checking cluster 'test-cluster' status.
Caused by: java.io.IOException: 408 Request Timeout
	at WaitForHttpResource.checkResource / waitFor
	at DefaultLocalClusterHandle.waitUntilReady / start / restart
```
The test performs a full cluster restart (`cluster.restart(false)`) and then waits for the
2-node cluster to come back up. `DefaultLocalClusterHandle` gives the cluster a fixed
`CLUSTER_UP_TIMEOUT` of 5 minutes to respond to an HTTP health probe; on this occasion the
node(s) apparently did not become reachable within that window and the probe returned a
408 (request timed out at the connection level), causing the wait to fail and the test to
throw. This looks like CI-host resource contention / slow node startup rather than a logic
bug in the test or in the restart/allocation code path — no other evidence (mapping/data
loss, wrong error, etc.) is present in the failure.

**Release risk: low-risk**

> The failure is a timeout waiting for a node HTTP endpoint to become ready after a full
> cluster restart in a QA test harness (`test-cluster` fixture code), not a production code
> path. It reproduces at a low, intermittent rate (0.6% of executions for the class, 2.5% for
> the step) across periodic/PR pipelines, consistent with CI host resource contention rather
> than a deterministic defect. A direct reproduction attempt with the same seed/locale/
> timezone/commit passed cleanly, further supporting a transient CI-environment cause. This
> would be raised to medium-risk if the failure rate increased sharply or if repeated
> reproductions with the exact seed also failed with the same 408 error.

**Impact**
- Affected branches: main
- Affected OS: ubuntu (linux x64), almalinux-8-aarch64 (arm) — periodic, periodic-platform-support, and pull-request pipelines
- Affected pipelines: elasticsearch-periodic, elasticsearch-periodic-platform-support, elasticsearch-pull-request
- Trend: intermittent, low rate (0.6% class-level, 2.5% step-level, 1.7% pipeline-level) — consistent with a CI-environment timing flake, not a regression tied to a specific change

**Reproduction**
- Command: `./gradlew ":x-pack:qa:repository-old-versions:yamlRestTest#5_6_16" --tests "org.elasticsearch.oldrepos.OldMappingsIT.testSurvivesRestart" -Dtests.seed=54147665F5F91BA2 -Dtests.locale=kk-Cyrl-KZ -Dtests.timezone=Indian/Reunion -Druntime.java=26`
- Commit: `e92ae9d943f23af69eef2051d3935db735c9a1cd`
- Remote job: https://buildkite.com/elastic/elasticsearch-agentic-workflow/builds/374#019fc56f-a862-486c-92e0-e64a2aa58adc — ubuntu-2404
- Result: **not reproduced** — test passed cleanly (1 test completed, BUILD SUCCESSFUL) using the exact seed/locale/timezone/commit from the original failing build
- Build scan: https://gradle-enterprise.elastic.co/s/ei5ki6l2f22pq

**Mitigation Proposal**
- Reproduction verdict: The clean pass with the identical seed lowers confidence in any
  deterministic/code-level root cause and raises confidence that this is a transient,
  host-load-dependent timeout during cluster restart.
- Codebase investigation: Reviewed `OldMappingsIT.testSurvivesRestart` (x-pack/qa/repository-old-versions),
  `DefaultLocalClusterHandle.waitUntilReady`/`restart` and `WaitForHttpResource.waitFor`/`checkResource`
  in `test/test-clusters`. `CLUSTER_UP_TIMEOUT` is a fixed `Duration.ofMinutes(5)` (bumped from a
  shorter value back to 5 minutes previously, per git history), applied uniformly regardless of
  cluster size/distribution/host load. No test- or fixture-specific code change looks implicated;
  git history on the test file shows two recent unrelated refactors (#155550, #154518) that moved
  to the unit-rule-based test cluster / testcontainer fixture, but nothing suggesting a functional
  regression in restart handling.
- Recent git history: #154518 (fixture migration to rule-based cluster/testcontainer) and #155550
  (isolate per-class repo dirs) both recently touched this test area; neither obviously explains a
  timeout-only failure, and the reproduction passing on the current HEAD is consistent with these
  changes not being the cause.
- Proposed fix: No code fix indicated. If flakiness persists at a rate that becomes disruptive:
  (1) consider a slightly longer or host-load-scaled timeout for full-cluster-restart waits in
  `DefaultLocalClusterHandle`/`WaitForHttpResource` used by 2-node clusters restarting via
  `usesDefaultDistribution`, and/or (2) add more diagnostic logging (e.g. node process state, port
  status) on `waitUntilReady` timeout to make any future recurrence faster to triage. This is
  test-infrastructure hardening, not a production fix.
- Effort estimate: **trivial** — if any action is taken, it would be a config/timeout tweak or added logging, not a functional change.
- Confidence: 0.55 (medium) — no reproduction achieved and no code defect found; conclusion rests on absence of contrary evidence plus a very low, intermittent failure rate typical of CI timing flakes.

--- a/x-pack/qa/repository-old-versions/src/yamlRestTest/java/org/elasticsearch/oldrepos/DocValueOnlyFieldsIT.java
+++ b/x-pack/qa/repository-old-versions/src/yamlRestTest/java/org/elasticsearch/oldrepos/DocValueOnlyFieldsIT.java
@@ -77,13 +77,27 @@ public class DocValueOnlyFieldsIT extends ESClientYamlSuiteTestCase {
     @Override
     protected Settings restClientSettings() {
         String token = basicAuthHeaderValue("admin", new SecureString("admin-password".toCharArray()));
-        return Settings.builder().put(ThreadContext.PREFIX + ".Authorization", token).build();
+        return Settings.builder()
+            .put(ThreadContext.PREFIX + ".Authorization", token)
+            // increase the socket timeout so it doesn't fire before the increased ensureGreen timeout below;
+            // waiting on cluster health after restoring/recovering the archived index can take longer than
+            // the default 30s/60s on busy/contended CI hosts (see ParameterizedRollingUpgradeTestCase).
+            .put(CLIENT_SOCKET_TIMEOUT, "90s")
+            .build();
     }
 
     @Override
     protected boolean skipSetupSections() {
         // setup in the YAML file is replaced by the method below
         return true;
+    }
+
+    @Override
+    protected String getEnsureGreenTimeout() {
+        // restoring a snapshot into a fresh cluster (and, after a restart, recovering it from disk) can take
+        // longer than the default 30s on busy/contended CI hosts; align with the timeout used by other
+        // BWC-style tests that wait on cluster health (see ParameterizedRollingUpgradeTestCase).
+        return "70s";
     }
 
     @Before

--- a/x-pack/qa/repository-old-versions/src/yamlRestTest/java/org/elasticsearch/oldrepos/OldMappingsIT.java
+++ b/x-pack/qa/repository-old-versions/src/yamlRestTest/java/org/elasticsearch/oldrepos/OldMappingsIT.java
@@ -99,7 +99,21 @@ public class OldMappingsIT extends ESRestTestCase {
     @Override
     protected Settings restClientSettings() {
         String token = basicAuthHeaderValue("admin", new SecureString("admin-password".toCharArray()));
-        return Settings.builder().put(ThreadContext.PREFIX + ".Authorization", token).build();
+        return Settings.builder()
+            .put(ThreadContext.PREFIX + ".Authorization", token)
+            // increase the socket timeout so it doesn't fire before the increased ensureGreen timeout below;
+            // waiting on cluster health after restoring/recovering the archived indices can take longer than
+            // the default 30s/60s on busy/contended CI hosts (see ParameterizedRollingUpgradeTestCase).
+            .put(CLIENT_SOCKET_TIMEOUT, "90s")
+            .build();
+    }
+
+    @Override
+    protected String getEnsureGreenTimeout() {
+        // restoring a snapshot into a fresh cluster (and, after a restart, recovering it from disk) can take
+        // longer than the default 30s on busy/contended CI hosts; align with the timeout used by other
+        // BWC-style tests that wait on cluster health (see ParameterizedRollingUpgradeTestCase).
+        return "70s";
     }
 
     @Before

